src/main/java/org/mqttbee/mqtt/handler/ssl/KeystoreUtil.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 The MQTT Bee project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mqttbee.mqtt.handler.ssl;
```

Comment:
This util is not used inside our library, isn't it?
So it should be moved to org.mqttbee.api.util or something like that.

Reply:
This util is not used at all right now. Since the SSL config builder only allows to configure KeyStore Objects and not Key-/TrustManagerFactories.

I did some recherche on this topic (https://docs.oracle.com/javase/7/docs/technotes/guides/security/jsse/JSSERefGuide.html#DefaultKTMCust) and we should probably change our builder interface to accept Key-/TrustManagerFactories rather then KeyStores, because this is the more generic approach and will provider bigger flexibility to the users. That's probably also the reason why Netty is using Key-/TrustManagerFactories on its interfaces to configure SSL.
Of course, then we can provide this util on our API to the users for convenience to create Key-/TrustManagerFactories from KeyStore files.

If we stick to KeyStores on our API, the user must use `ssl.KeyManagerFactory.algorithm` and `ssl.TrustManagerFactory.algorithm` security properties to change the default algorithms (`SunX509` respectively `PKIX`) that are currently used by the builder to create the factories.

What do you think?

Reply:
I think it is best to accept Key-/TrustManagerFactories in the builder and remove the Keystore methods completely.
Maybe we can add more methods to the util sometime later.

Reply:
I changed the API of the builder. I also moved the KeyStoreUtil to org.mqttbee.api.util

The KeyStoreUtil is now used by the client example and the smoke test to create the Key-/TrustManagerFactories from files.